modules/renter/hostdb/hostdb.go

```diff
@@ -38,18 +65,51 @@ type HostDB struct {
 	// scan.
 	scanPool chan *hostEntry
 
-	mu *sync.RWMutex
+	mu sync.RWMutex
+}
+
+// a hostContract includes the original contract made with a host, along with
+// the most recent revision.
+type hostContract struct {
+	ID              types.FileContractID
+	FileContract    types.FileContract
+	LastRevisionTxn types.Transaction
+
+	// revisions must happen in serial
+	mu sync.RWMutex // TODO: consider `json:"-"`
```

Comment:
ah, that was to prevent `json.Marshal` from accessing the `mu` field, causing a data race...I haven't tested that it would even work as hoped, though, and there are more robust ways to handle that race anyway.